src/Typedefs/Typedefs.idr

```diff
@@ -345,3 +345,37 @@ mutual
 implementation Eq (TNamed n) where
   (TName n t) == (TName n' t')       = n == n' && t == t'
 
+-- Debug ----
+
+-- we use a named implementation of `Show (Fin n)` to avoid possible conflicts
+[finSimpleShow] Show (Fin k) where
+  show = show . finToNat
+
+mutual
+  debugTDefVect : Vect k (TDef n) -> String
```

Comment:
Why can't this be implemented with `show` on vectors?

Reply:
Probably because of `debugTDef` reference

Reply:
Probably because of `debugTDef`

Reply:
ah, this could actually be a named implementation of the `Show` interface, let me try if it works

Reply:
actually this does not seem to work easily because `show :: Vect n a -> String` expects an implementation of `Show (Vect n a)` and not an implementation of `Show a`

Reply:
yeah, you'll need to make everything named instances and thread them through

Reply:
ok, so I think this is ready to be merged now